modules/ephemeral-storage-sharing-secrets-across-namespaces.adoc

```diff
@@ -0,0 +1,33 @@
+:_mod-docs-content-type: PROCEDURE
+
+[id="ephemeral-storage-sharing-secrets-across-namespaces_{context}"]
+= Sharing secrets across namespaces
+
+[role="_abstract"]
+To share a secret across namespaces in a cluster, you create a `SharedSecret` custom resource (CR) instance for the `Secret` object that you want to share.
+
+.Prerequisites
+
+You must have permission to perform the following actions:
+
+* Create instances of the `sharedsecrets.sharedresource.openshift.io` custom resource definition (CRD) at a cluster-scoped level.
+* Create a ClusterRole for the SharedConfigMap.
+* Create a Role and RoleBinding for the Shared Resource CSI Driver itself.
+* Manage roles and role bindings across the namespaces in the cluster to control which users can get, list, and watch those instances.
+* Manage roles and role bindings to control whether the service account specified by a pod can mount a Container Storage Interface (CSI) volume that references the `Secret` or `Configmap` CR instance you want to use.
+* Access the namespaces that contain the secrets you want to share.
+
+.Procedure
+
+* Create a `Secret` or `Configmap` CR instance to share across the source namespace.
++
+[source,terminal]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: shared-config
+  namespace: <name_of_the_source_namespace>
+data:
+  config.txt: "Hello world!"
+----
```

Comment:
Need yaml codes for creating `SharedSecret` object, `ClusterRole` , and `Role/RoleBinding`.

Reply:
Maybe we change the name of the `secret` because its misleading.

```
---
apiVersion: v1
kind: Secret
metadata:
  name: test-secret
  namespace: <source-namespace>
stringData:
  key1: "secret-data"
---
apiVersion: sharedresource.openshift.io/v1alpha1
kind: SharedSecret
metadata:
  name: shared-test-secret
spec:
  secretRef:
    name: test-secret
    namespace: <app-namespace>
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: use-shared-test-secret
rules:
  - apiGroups:
      - sharedresource.openshift.io
    resources:
      - sharedsecrets
    resourceNames:
      - shared-test-secret
    verbs:
      - use
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: use-shared-test-secret
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: use-shared-test-secret
subjects:
  - kind: ServiceAccount
    name: pipeline
  - kind: ServiceAccount
    name: builder
---
```

Reply:
Thank you @sayan-biswas